Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkCardComponent.swift

```diff
@@ -53,50 +54,40 @@ public class NolPayLinkCardComponent: PrimerHeadlessCollectDataComponent {
             self.mobileNumber = mobileNumber
-            self.phoneCountryDiallingCode = phoneCountryDiallingCode
         case .otpData(let otpCode):
             self.otpCode = otpCode
         }
         
-        // Notify validation delegate after updating data
-        let validations = validateData(for: collectableData)
-        validationDelegate?.didValidate(validations: validations, for: collectableData)
+        validateData(for: collectableData)
     }
     
-    func validateData(for data: NolPayLinkCollectableData) -> [PrimerValidationError] {
+    func validateData(for data: NolPayLinkCollectableData){
+        validationDelegate?.didUpdate(validationStatus: .validating, for: data)
         var errors: [PrimerValidationError] = []
         
         switch data {
             
-        case .phoneData(mobileNumber: let mobileNumber,
-                        phoneCountryDiallingCode: let phoneCountryDiallingCode):
-            if !mobileNumber.isValidMobilePhoneNumber {
-                errors.append(PrimerValidationError.invalidPhoneNumber(
-                    message: "Phone number is not valid.",
-                    userInfo: [
-                        "file": #file,
-                        "class": "\(Self.self)",
-                        "function": #function,
-                        "line": "\(#line)"
-                    ],
-                    diagnosticsId: UUID().uuidString))
-                ErrorHandler.handle(error: errors.last!)
-            }
-            
-            if !phoneCountryDiallingCode.isValidCountryCode {
-                errors.append(PrimerValidationError.invalidPhoneNumberCountryCode(
-                    message: "Country code is not valid.",
-                    userInfo: [
-                        "file": #file,
-                        "class": "\(Self.self)",
-                        "function": #function,
-                        "line": "\(#line)"
-                    ],
-                    diagnosticsId: UUID().uuidString))
-                ErrorHandler.handle(error: errors.last!)
-                
+        case .phoneData(mobileNumber: let mobileNumber):
+            phoneMetadataService.getPhoneMetadata(mobileNumber: mobileNumber) { [weak self] result in
+                switch result {
+                    
+                case let .success((validationStatus, countryCode, mobileNumber)):
+                    switch validationStatus {
+                        
+                    case .valid:
+                        self?.countryCode = countryCode
+                        self?.mobileNumber = mobileNumber
+                        self?.validationDelegate?.didUpdate(validationStatus: .valid, for: data)
+                    case .invalid(errors: let validationErrors):
+                        errors += validationErrors
+                        self?.validationDelegate?.didUpdate(validationStatus: .invalid(errors: errors), for: data)
+                    default: break
```

Comment:
Is it okay from the merchant perspective to get `.validating` but not receive a follow up e.g.`.validationCancelled` or `.error(.unknown)` - something like that?

I'm thinking of this happening where a merchant shows a spinner or something while waiting for validation, but never gets a callback (this is also a problem for co-badged cards).

Reply:
What I mean is, if the case is `PrimerValidationStatus.error` here, the delegate won't be called at all, because it will hit the `default: break` case

Reply:
That case is not possible, read `NolPayPhoneMetadataService.getPhoneMetadata(mobileNumber: String, completion: @escaping PhoneMetadataCompletion)` function. .error is only in the case when ApiCall fails, and in that case, we are returning that error (`case error(error: PrimerError)`). Maybe I don't see something that you see?

Reply:
Ah I think I get it - so it's filtering it into an error result at the level above? If so then this is fine 👍